src/frontend/magnify/src/i18n/TranslationProvider/TranslationsProvider.tsx

```diff
@@ -9,31 +12,88 @@ export interface TranslationProviderProps {
   /**
    * The current locale
    */
-  locale: string;
+  locale?: string;
   /**
    * The default locale
    */
   defaultLocale: string;
-  /**
-   * Messages for the current locale
-   *
-   * @see loadLocaleData to fetch messages for a given locale for this lib
-   * You may need to merge it with your own project messages
-   */
-  messages: Record<string, string> | Record<string, MessageFormatElement[]> | undefined;
+
+  initTranslation?: boolean;
+}
+
+async function getTranslation(
+  locale: string,
+): Promise<Maybe<Record<string, string> | Record<string, MessageFormatElement[]>>> {
+  let translatedMessages: Maybe<Record<string, string> | Record<string, MessageFormatElement[]>>;
+  try {
+    translatedMessages = await loadLocaleData(locale);
+  } catch (error) {
+    translatedMessages = {};
+  }
+  return translatedMessages;
+}
+
+export interface LocaleContextInterface {
+  currentLocale: string;
+  setCurrentLocale: (locale: string) => void;
 }
 
+const LocaleContext = React.createContext<Maybe<LocaleContextInterface>>(undefined);
+
 export default function TranslationProvider({
   children,
-  locale,
+  locale = MagnifyLocales.FR,
   defaultLocale,
```

Comment:
Maybe defaultLocale should have also a default value declared here ? `MagnifyLocales.EN` ?